### PR TITLE
use ANY operator instead of IN internally

### PR DESCRIPTION
**Pull request by mfelsche**
to support more than 1k elements.

And extended usage of the ANY operator to query for array literal ( column = ANY ([1,2,3]) ).

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -439,9 +439,44 @@ protected Symbol visitInPredicate(InPredicate node, ExpressionAnalysisContext co
...
-            InListExpression listExpression = ((InListExpression) node.getValueList());
+            List<Expression> expressions = ((InListExpression) node.getValueList()).getValues();
 
-            Set<Function> comparisons = new HashSet<>(listExpression.getValues().size());
+            boolean useAny = true;
+            Set<Literal> literals = new HashSet<>(expressions.size());
+            for (Expression expression : expressions) {
+                Symbol arrayElement = process(expression, context);
+                Literal literal;
+                if (arrayElement instanceof Literal) {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java]: use arrayElement.symbolType() instead

File: sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -439,9 +439,44 @@ protected Symbol visitInPredicate(InPredicate node, ExpressionAnalysisContext co
...
+                    literal = (Literal) arrayElement;
+                } else {
+                    Symbol normalized = normalize(arrayElement);
+                    if (normalized instanceof Literal) {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java]: here same, avoid instanceof

File: sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java [hunk 3]
```diff
@@ -761,11 +909,10 @@ private static GeoPointFieldMapper getGeoPointFieldMapper(String fieldName, Sear
...
-        private static final LteQuery lteQuery = new LteQuery();
-        private static final GtQuery gtQuery = new GtQuery();
-        private static final GteQuery gteQuery = new GteQuery();
-        private static final LikeQuery likeQuery = new LikeQuery();
+        private static final RangeQuery ltQuery = new RangeQuery("lt");
+        private static final RangeQuery lteQuery = new RangeQuery("lte");
+        private static final RangeQuery gtQuery = new RangeQuery("gt");
+        private static final RangeQuery gteQuery = new RangeQuery("gte");
         private static final WithinQuery withinQuery = new WithinQuery();
         private final ImmutableMap<String, FunctionToQuery> functions =
```
Comment (seut) [inline sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java]: can't this be also static?

File: sql/src/test/java/io/crate/analyze/where/EqualityExtractorTest.java [hunk 4]
```diff
@@ -333,4 +346,62 @@ public void testPK2_Eq1() throws Exception {
         assertNull(matches);
     }
 
+    @Test
+    public void testPK1_AnyEq() throws Exception {
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/where/EqualityExtractorTest.java]: hm not so meaningful method names, more verbose wouldn't hurt ;) also all other tests in this class

**Comment by seut** [comment]
otherwise ack

File: sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java (hunk 3 above)
Comment (mfelsche) [inline sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java]: no, as AndQuery, OrQuery and NotQuery are not static :(

**Pull request merged**